crates/block-producer/src/mempool/mod.rs

```diff
@@ -309,11 +309,15 @@ impl Mempool {
     /// The pool will mark the associated batches and transactions as committed, and prune stale
     /// committed data, and purge transactions that are now considered expired.
     ///
+    /// # Returns
+    ///
+    /// Returns the list of transactions that expired and were purged.
```

Comment:
nit: I would maybe make this a bit more general. For example: "Returns a set of transactions that were purged from the mempool because they can no longer be included in in the chain (e.g., expired transactions and their descendants)."